package/network/services/dnsmasq/files/dnsmasqsec.hotplug

```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+
+TIMEVALIDFILE=/tmp/ntp.d/dnsmasqsec
+
+[ "$ACTION" = stratum ] || exit 0
```

Comment:
You might check for either stratum or step action. When I look at my own system log where I have enabled ntpd event logging, the step event comes first (when the time is initially set) and then a bit later the first stratum event to tell that the quality of the ntpd time has changed. However, there may be more than 30 seconds between the initial time set event and the first stratum event. It might be useful to restart dnsmasq already earlier.

Log extract from the last boot: 
```
root@lede:~# logread | grep ntpd
Thu Jun 16 11:01:40 2016 user.notice ntpd: Time set, stratum=16 interval=32 offset=3593.824949
Thu Jun 16 11:02:12 2016 user.notice ntpd: Stratum change, stratum=2 interval=32 offset=0.000069
```